pkgs/applications/misc/ranger/default.nix

```diff
@@ -1,4 +1,7 @@
-{ stdenv, fetchurl, pythonPackages, w3m, file, less }:
+{ stdenv, fetchurl, pythonPackages, file, less
+, imagePreviewSupport ? true, w3m ? null}:
```

Comment:
I wonder what the purpose of defaulting w3m to `null` is? As far as I can tell, the expression fails to evaluate if `w3m = null` is in fact passed here (provided `imagePreviewSupport = true`). That'd require user intervention of course, but still seems odd to default to an invalid combination of parameter values.